apps/trading/e2e/tests/referrals/test_referrals.py

```diff
@@ -0,0 +1,131 @@
+import re
+import pytest
+from playwright.sync_api import Page, expect
+from vega_sim.service import VegaService
+from conftest import init_vega
+from fixtures.market import setup_continuous_market, setup_simple_market
+from actions.utils import WalletConfig, change_keys, create_and_faucet_wallet, element_contains_text, next_epoch
+from actions.vega import submit_order, submit_liquidity
+from wallet_config import MM_WALLET, MM_WALLET2, TERMINATE_WALLET
+
+PARTY_A = WalletConfig("party_a", "party_a")
+PARTY_B = WalletConfig("party_b", "party_b")
+
+@pytest.fixture(scope="module")
+def vega(request):
+    with init_vega(request) as vega:
+        yield vega
+
+sell_orders = [[1, 111], [1, 111], [1, 112], [1, 112], [1, 113], [1, 113], [1, 114], [1, 114], [1, 115], [1, 115]]
+buy_orders = [[1, 106], [1, 107], [1, 108]]
+
+@pytest.fixture(scope="module")
+def continuous_market(vega):
+    market = setup_simple_market(vega, custom_quantum=100000)
+    return setup_continuous_market(vega, market, buy_orders, sell_orders, add_liquidity=False)
+
+def check_volume_and_tier(page: Page, expected_volume, expected_tier):
+    if "referrals" in page.url:
+         page.reload()
+    else:
+         page.goto("/#/referrals/")
+
+    combined_volume_element = page.get_by_test_id('combined-volume-value')
+    current_tier_element = page.get_by_test_id('current-tier-value')
+
+    volume_matched = element_contains_text(combined_volume_element, expected_volume)
+    tier_matched = element_contains_text(current_tier_element, expected_tier)
+
+    return volume_matched, tier_matched
+
+@pytest.mark.usefixtures("page", "auth", "risk_accepted")
+def test_referral_scenario(continuous_market, vega: VegaService, page: Page):
+    page.goto(f"/#/markets/{continuous_market}")
```

Comment:
I would move the page goto, after doing everything vega side, just so it's clear when UI testing has started.